firefox-webext/background.js

```diff
@@ -17,3 +106,59 @@ chrome.browserAction.onClicked.addListener(function(tab) {
     });
   });
 });
+
+function storeIssues() {
+  chrome.storage.local.set({issuesList, tsRefresh}, function(){
+    if (chrome.runtime.lastError) {
+      console.log(chrome.runtime.lastError);
+    } else {
+      console.log("Local store complete.");
+    }
+  });
+}
+
+function storageCheck(item){
+  if (chrome.runtime.lastError) {
+    console.log(chrome.runtime.lastError);
+  } else {
+    if (Object.keys(item).length === 0 && item.constructor === Object){
+      getIssues(); // if nothing in storage fetch the list from github
+    } else {
+      issuesList = item.issuesList;
+      tsRefresh = item.tsRefresh;
+      console.log('Issues loaded from storage.');
+      checkRefresh();
+    }
+  }
+}
+
+function extractDomain(url) {
```

Comment:
Another way to do this is to let the DOM do the work:

```
function getDomain(url) {
  var f = document.createElement('a')
  f.href = url;
  return f.hostname;
```

And you also have access to `f.host` & `f.hostname` (blah.com), `f.origin` (http://blah.com) and `f.href` (the original URL).